src/upload/src/Upload.tsx

```diff
@@ -293,6 +302,15 @@ export default defineComponent({
       // May have bug! set to null?
       target.value = ''
     }
+    function handleFileListChange (): void {
+      const fileListAfterChange = mergedFileListRef.value
+       const {
+          'onUpdate:fileList': _onUpdateFileList,
+          onUpdateFileList,
+        } = props
+        if (_onUpdateFileList) call(_onUpdateFileList, fileListAfterChange)
+        if (onUpdateFileList) call(onUpdateFileList, fileListAfterChange)
+    }
```

Comment:
这个不用 inject 吧？直接在 doChange 里面调用就行了。

Reply:
```suggestion
    function doUpdateFileList (value: Xxx): void {
       const {
          'onUpdate:fileList': _onUpdateFileList,
          onUpdateFileList,
        } = props
        if (_onUpdateFileList) call(_onUpdateFileList, value)
        if (onUpdateFileList) call(onUpdateFileList, value)
        uncontrolledFileListRef.value = value
    }
```

然后注意第 400 行，改成调用 doUpdateFileList()

Reply:
> 这个不用 inject 吧？直接在 doChange 里面调用就行了。

在dochange里面调用会造成多次触发，实际上我们只要filelist改变时触发就可以，状态改变不需触发

Reply:
没看出来会重复，file 变了那 fileList 一定会变

Reply:
好像是我理解错了,add load progress时都触发是合理的是吧